### PR TITLE
Allow to build windows binaries

### PR DESCRIPTION
**Pull request by MarcosDY**
Solve required changes on code to be able to build helper on windows,
this changes does not allow to create artifacts or run builds on ci

### CODE REVIEW

File: pkg/sidecar/util_windows.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	return nil
+}
+
+func (s *Sidecar) SignalProcess() error {
+	// Signal to reload certs
+	// TODO: it is not possible to get signal by name on windows,
+	// we must provide int here
+	return errors.New("sending signal is not supported on windows")
+}
```
Comment (faisal-memon) [inline pkg/sidecar/util_windows.go:31]: So the process can't be reloaded on windows?

Comment (MarcosDY) [reply pkg/sidecar/util_windows.go:31]: windows is different to linux, there is no support for `Signal` like on linux,
and only support 
```
	Interrupt Signal = syscall.SIGINT
	Kill      Signal = syscall.SIGKILL
```

So generally what apps are doing on windows is reloading based on another kind of communications, 
like sql scripts or open APIs

Comment (MarcosDY) [reply pkg/sidecar/util_windows.go:31]: so, for reloading people can still use run script approach

File: Makefile [hunk 2]
```diff
@@ -76,11 +79,18 @@ build_dir := $(DIR)/.build/$(os1)-$(arch1)
...
 go_version := $(shell cat .go-version)
 go_dir := $(build_dir)/go/$(go_version)
-go_bin_dir := $(go_dir)/bin
-go_url = https://storage.googleapis.com/golang/go$(go_version).$(os1)-$(arch2).tar.gz
-go := PATH="$(go_bin_dir):$(PATH)" go
+ifeq ($(os1),windows)
+	go_bin_dir = $(go_dir)/go/bin
+	go_url = https://storage.googleapis.com/golang/go$(go_version).$(os1)-$(arch2).zip
+	exe=".exe"
+else
```
[the remark this answers is not in the document]
Comment (MarcosDY) [reply Makefile:86]: mmm, but we build macos with this lines too, 
it is only special case for windows that uses zip to upload go,
and where we need to change build output

File: pkg/sidecar/util_windows.go (hunk 1 above)
Comment (faisal-memon) [reply pkg/sidecar/util_windows.go:31]: Should we document this?

Comment (amartinezfayo) [reply pkg/sidecar/util_windows.go:31]: I was thinking about the same. In addition to document that signaling is not supported on Windows, I think that ideally we should check if the configuration has the `renewSignal` setting when running on Windows and fail indicating that is not supported.

File: README.md [hunk 3]
```diff
@@ -6,29 +6,29 @@
...
 
 If `-config` is not specified, the default value `helper.conf` is assumed. 
 
-### Configuration
+## Configuration
 The configuration file is an [HCL](https://github.com/hashicorp/hcl) formatted file that defines the following configurations:
 
  |Configuration        | Description                                                                                    | Example Value |
  |--------------------------|------------------------------------------------------------------------------------------------| ------------- |
- |`agentAddress`            | Socket address of SPIRE Agent.                                                                 | `"/tmp/agent.sock"`                                                                                                                                                  |
```
Comment (faisal-memon) [inline README.md:21]: Need to remove "Example Value" column above.

**Review by faisal-memon — APPROVED** [review]

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**